common/src/main/java/org/apache/rocketmq/common/message/MessageDecoder.java

```diff
@@ -163,7 +166,8 @@ public static byte[] encode(MessageExt messageExt, boolean needCompress) throws
         int storehostAddressLength = (sysFlag & MessageSysFlag.STOREHOSTADDRESS_V6_FLAG) == 0 ? 8 : 20;
         byte[] newBody = messageExt.getBody();
         if (needCompress && (sysFlag & MessageSysFlag.COMPRESSED_FLAG) == MessageSysFlag.COMPRESSED_FLAG) {
-            newBody = UtilAll.compress(body, 5);
+            Compressor compressor = CompressorFactory.getCompressor(MessageSysFlag.getCompressionType(sysFlag));
+            newBody = compressor.compress(body, 5);
```

Comment:
Do we need to get compress level from `System.getProperty(MixAll.MESSAGE_COMPRESS_LEVEL, "5")`?

Reply:
I don't know why it was 5 at the first, so I think it's better to keep things as they are, and avoid any compatibility issues.